src/features/defi/components/Deposit/Deposit.tsx

```diff
@@ -133,12 +133,18 @@ export const Deposit = ({
 
   const handlePercentClick = useCallback(
     (percent: number) => {
-      const cryptoAmount = bnOrZero(cryptoAmountAvailable).times(percent).precision(asset.precision)
-      const fiatAmount = bnOrZero(cryptoAmount).times(marketData.price)
-      setValue(Field.FiatAmount, fiatAmount.toString(), {
+      // The human crypto amount as a result of amount * percentage / 100, possibly with too many digits
+      const percentageCryptoAmount = bnOrZero(cryptoAmountAvailable).times(percent)
+      const percentageFiatAmount = percentageCryptoAmount.times(marketData.price)
+      const percentageCryptoAmountHuman = percentageCryptoAmount
+        .decimalPlaces(asset.precision)
```

Comment:
Seems like we were using `precision` as a built-in replacement for `decimalPlaces`. `precision` was a no-op in this case